src/test/groovy/unit/org/pih/warehouse/product/ProductControllerTests.groovy

```diff
@@ -1,151 +1,46 @@
-				"\"1234\",\"product 1234\",\"category 123\",\"\",\"\",\"\",\"\",\"\",\"false\",\"\",\"\",\"${dateString}\",\"${dateString}\"\n"
-				"\"1236\",\"product 1235\",\"category 123\",\"\",\"\",\"\",\"\",\"\",\"false\",\"\",\"\",\"${dateString}\",\"${dateString}\""
-		}
-		controller.productService = productServiceMock.createMock()
-		
-		
-        //def model = controller.exportAsCsv()
-		//println controller.response
-		//println controller.response.contentAsString
-		
-		//def lines = controller.response.contentAsString.split("\n")
-		//assertEquals '"ID","Name","Category","Description","Product Code","Unit of Measure","Manufacturer","Manufacturer Code","Cold Chain","UPC","NDC","Date Created","Date Updated"', lines[0]
-		//assertEquals '"1234","product 1234","category 123","","","","","","false","","","${dateString}","${dateString}"', lines[1]
-		//assertEquals '"1236","product 1236","category 123","","","","","","false","","","${dateString}","${dateString}"', lines[2]
-    }
-	
-	/*
-	void test_import_shouldCreateNewProduct() {
-		def csv = """\"ID\",\"Name\",\"Category\",\"Description\",\"Product Code\",\"Unit of Measure\",\"Manufacturer\",\"Manufacturer Code\",\"Cold Chain\",\"UPC\",\"NDC\",\"Date Created\",\"Date Updated\"\n\"1235\",\"product 1235\",\"category 123\",\"\",\"\",\"\",\"\",\"\",\"false\",\"\",\"\",\"\",\"\""""
-		println csv;		
+		def category = new Category(id: "123", name: "category 123").save(flush: true)
+		new Category(id: "123", name: "category 123").save(flush: true)
+		new Product(id:"1234", name: "product 1234", category: category, lastUpdated: currentDate, dateCreated: currentDate).save(flush: true)
+		new Product(id:"1236", name: "product 1236", category: category, lastUpdated: currentDate, dateCreated: currentDate).save(flush: true)
 	}
 
-	void test_import_shouldUpdateExistingProduct() {		
-		def csv = """\"ID\",\"Name\",\"Category\",\"Description\",\"Product Code\",\"Unit of Measure\",\"Manufacturer\",\"Manufacturer Code\",\"Cold Chain\",\"UPC\",\"NDC\",\"Date Created\",\"Date Updated\"\n\"1236\",\"product 1236\",\"category 123\",\"\",\"\",\"\",\"\",\"\",\"false\",\"\",\"\",\"\",\"\""""		
-		println csv;
+	void "test product list size"() {
+		expect:
+		Product.count() == 2
 	}
-	*/
 
-	/*
-	void testSave() {
-		def category1 = new Category(id: "root", name: "root")
-		def category2 = new Category(id: "123", name: "category 123")
-		def product1 = new Product(id:"1234", name: "product 1234", category: category2)
-		def product2 = new Product(id:"1236", name: "product 1236", category: category2)
-		def location1 = new Location(id: "1", name: "Boston Headquarters")
-		mockDomain(Location, [location1])
-		mockDomain(Product, [product1, product2])
-		mockDomain(Category, [category1, category2])
-		mockDomain(Attribute)
-		
-		def productControl = mockFor(ProductService)
-		productControl.demand.getRootCategory(1..1) { return category1 }
+	void "test list should contain two products"() {
```

Comment:
For most tests, I think "should contain two products" will suffice. I know you just copied this from the original test name. So leave it for now build but we should come up with our own convention and/or try to see if there's a more BDD convention.

Reply:
Here's a good article describing BDD that I wholeheartedly agree with https://dannorth.net/introducing-bdd/ (referenced from https://dannorth.net/whats-in-a-story/).

Reply:
Yup you're right, for old tests I just left the original test name, but for new I was basing on the "test something" from docs. I'll review my test names and decide which version is better.

Reply:
fixed